### PR TITLE
Pass message to error handler

### PR DESCRIPTION
**Pull request by briancavalier**
The main purpose of this PR is to pass both the Error and AMQP Message to error handlers when using defaultParseAndHandleMessage.

### Motivation

This helps in logging more useful information about failed messages, since many times, the Error object contains only a generic description and stack trace.  For example, in many of our use cases, it would be useful to be able to log additional info, like the associated businessId or visitId, which probably only exists in the message payload.

### Other changes

This PR also:

* Updates all deps and devDeps, including refactoring to appease newer jsinspect strictness
* Switches from isparta to [nyc](https://github.com/istanbuljs/nyc).  [Isparta is no longer maintained](https://github.com/douglasduteil/isparta#isparta---), and nyc is the new official front end to istanbul code coverage.  The [license is ISC](https://github.com/istanbuljs/nyc/blob/master/LICENSE.txt), which is functionally equivalent to MIT.
* Tightens types on a few things, including error handlers.

### CODE REVIEW

**Comment by briancavalier** [comment]
This PR should also help with #9, since it updates to a much newer version of eslint.

File: .gitignore
Comment (ddelnano) [inline .gitignore]: @briancavalier should we make this recursive just cause?

Comment (briancavalier) [inline .gitignore]: Sure

Comment (briancavalier) [inline .gitignore]: Done in f94fb93

File: test/defaultParseAndHandleMessage-test.js
Comment (raychan) [inline test/defaultParseAndHandleMessage-test.js]:
```
const error = new Error();
const handleError = async (err) => {
      assert.strictEqual(error, err)
}

const handleMessage = async (data) => {
      assert.equal(content, JSON.stringify(data))
      throw error
}
```

**Review by raychan — CHANGES_REQUESTED** [review]
Few minor comments

**Comment by ddelnano** [comment]
Took me a while to wrap my head around this since I haven't look at this functional JS in a while but aside from @raychan's comment (which I didn't think about when I got my head wrapped around things) lgtm

**Comment by ddelnano** [comment]
And ❤️ that this should fix #9.  After we merge this @briancavalier can you rebase #9 and see if that fixes things?

File: test/defaultParseAndHandleMessage-test.js
Comment (briancavalier) [inline test/defaultParseAndHandleMessage-test.js]: :+1:

File: test/defaultParseAndHandleMessage-test.js
Comment (briancavalier) [inline test/defaultParseAndHandleMessage-test.js]: I think it's sufficient to assert [here](https://github.com/nowait/amqp/blob/d72ef659b56b7db24c6ef886258287b835cac0f2/test/defaultParseAndHandleMessage-test.js#L30).

File: test/defaultParseAndHandleMessage-test.js
Comment (briancavalier) [inline test/defaultParseAndHandleMessage-test.js]: Good call.

**Comment by briancavalier** [comment]
FWIW, this is an API-breaking change (existing `handleError` functions passed to `defaultParseAndHandleMessage` won't type check), so it'll require a major version bump when we release it.

**Comment by briancavalier** [comment]
@ray Addressed your feedback in 4d27dc7

**Review by raychan — APPROVED** [review]

**Comment by raychan** [comment]
LGTM

**Comment by briancavalier** [comment]
I'll merge this and publish a major version bump.

**Pull request merged**
